packages/drivers/routerlicious-driver/src/restWrapper.ts

```diff
@@ -131,19 +142,37 @@ export class RouterliciousRestWrapper extends RestWrapper {
 		statusCode: number,
 		canRetry = true,
 	): Promise<IR11sResponse<T>> {
+		// Check whether this request has been made before or if it is a retry.
+		// requestKey is built using the HTTP method appended with the complete URL ommitting the 'retry' param
+		const url = new URL(buildRequestUrl(requestConfig));
+		url.searchParams.delete("retry");
```

Comment:
Why don't we fetch the retry count from the url, and just increment it there; as opposed to first deleting it and then updating a local counter and inserting it in the URL?

Reply:
@rajatch-ff / @znewton The logic would break if the `retry` parameter is provided by the caller of this function. So here we ensure that the retry param is always picked from the `retryCounter` map which we maintain here locally.